spec/defines/fastcgi_server_spec.rb

```diff
@@ -8,162 +8,111 @@
     'www'
   end
 
-  describe 'os-dependent items' do
-    context 'on RedHat based systems' do
-      let :default_facts do
-        {
-          osfamily: 'RedHat',
-          operatingsystem: 'CentOS',
-          operatingsystemrelease: '6',
-          kernel: 'Linux',
-          id: 'root',
-          path: '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
-          is_pe: false,
-        }
+  on_supported_os.each do |os, facts|
+    next if facts[:os]['release']['major'] == '18.04'
+    next if facts[:os]['release']['major'] == '7' && facts[:os]['family']['RedHat']
+    context "on #{os} " do
+      let :facts do
+        facts
       end
-      let(:facts) { default_facts }
 
-      it { is_expected.to contain_class('apache') }
-      it { is_expected.to contain_class('apache::mod::fastcgi') }
-      it {
-        is_expected.to contain_file("fastcgi-pool-#{title}.conf").with(
-          ensure: 'file',
-          path: "/etc/httpd/conf.d/fastcgi-pool-#{title}.conf",
-        )
-      }
-    end
-    context 'on Debian based systems' do
-      let :default_facts do
-        {
-          osfamily: 'Debian',
-          operatingsystem: 'Debian',
-          operatingsystemrelease: '8',
-          lsbdistcodename: 'jessie',
-          kernel: 'Linux',
-          id: 'root',
-          path: '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
-          is_pe: false,
-        }
+      case facts[:os]['family']
+      when 'RedHat'
+        context 'on RedHat based systems' do
+          it { is_expected.to contain_class('apache') }
```

Comment:
This expectation can be pulled out of the case statement. An alternative is to put the variable `path` assignment in a case statement and have the same expectations for all.

Reply:
Top two it's have been moved outside the case